lib/command/src/Obelisk/Command/Thunk.hs

```diff
@@ -136,6 +129,45 @@ data GitSource = GitSource
   }
   deriving (Show, Eq, Ord)
 
+makeLensesFor [("_gitSource_url", "gitSource_url")] ''GitSource
+
+-- | A location from which a thunk's data can be retrieved
+data ThunkSource
+   -- | A source specialized for GitHub
+   = ThunkSource_GitHub GitHubSource
+   -- | A plain repo source
+   | ThunkSource_Git GitSource
+   deriving (Show, Eq, Ord)
+
+makePrisms ''ThunkSource
+
+-- | A reference to the exact data that a thunk should translate into
+data ThunkPtr = ThunkPtr
+  { _thunkPtr_rev :: ThunkRev
+  , _thunkPtr_source :: ThunkSource
+  }
+  deriving (Show, Eq, Ord)
+
+makeLenses 'ThunkPtr
+
+--TODO: Support symlinked thunk data
+data ThunkData
+   = ThunkData_Packed ThunkPtr
+   -- ^ Packed thunk
+   | ThunkData_Checkout (Maybe ThunkPtr)
+   -- ^ Checked out thunk that was unpacked from this pointer
+  deriving (Show, Eq, Ord)
+
+-- | Whether or not to override the scheme when packing/unpacking a thunk.
+data ThunkScheme
+  = ThunkScheme_Keep
+  -- ^ Keep the protocol https/ssh as it was.
```

Comment:
It's also possible for it to be `file://` (and `git://` which we don't support yet) but it doesn't make sense for you to switch to or from `file://` because that doesn't have the same URI format.